sklearn/neighbors/lof.py

```diff
@@ -250,26 +262,52 @@ def _predict(self, X=None):
         if X is not None:
             X = check_array(X, accept_sparse='csr')
             is_inlier = np.ones(X.shape[0], dtype=int)
-            is_inlier[self._decision_function(X) < 0] = -1
+            is_inlier[self.decision_function(X) < 0] = -1
         else:
             is_inlier = np.ones(self._fit_X.shape[0], dtype=int)
             is_inlier[self.negative_outlier_factor_ < self.offset_] = -1
 
         return is_inlier
 
-    def _decision_function(self, X):
+    def predict(self, X):
```

Comment:
Should we use a property to make this disappear when not novelty?

Reply:
I know the concept of properties but I'm not very familiar with it so it's not very clear to me how this would work. If you could clarify what you have in mind there is a good chance that I will be +1 to make this disappear when not novelty :).

Reply:
Thanks. The goal of doing this is to have `hasattr` return False when not novelty. This is what you meant when you wrote "make this disappear when not novelty", i.e. make predict disappear when not novelty. Right?

Reply:
In Python 3, `hasattr` returns False if an AttributeError is raised. This means that I need to raise an AttributeError if not novelty whereas we were raising a ValueError. That being said I'm also fine with AttributeError, as the existence of the attribute depends on the value of novelty.

Reply:
But I think an AttributeError is raised if `__get__` fails with whichever error. But I might not be recalling correctly. 

Yes, do it for the other methods too